frontend/ui/screensaver.lua

```diff
@@ -450,6 +450,29 @@ function Screensaver:setup(event, event_message)
                 doc_settings = DocSettings:open(lastfile)
             end
             excluded = doc_settings:isTrue("exclude_screensaver")
+
+            local book_summary = doc_settings:readSetting("summary")
+            local book_finished = book_summary and book_summary.status == "complete"
+            local book_on_hold = book_summary and book_summary.status == "abandoned"
+            local exclude_finished_books = G_reader_settings:isTrue("screensaver_exclude_finished_books") and book_finished
+            local exclude_on_hold_books = G_reader_settings:isTrue("screensaver_exclude_on_hold_books") and book_on_hold
+            local exclude_books_in_fm = not ui and G_reader_settings:isTrue("screensaver_hide_cover_in_filemanager")
+
+            if not excluded then
```

Comment:
These local variables are badly named:
`exclude_finished_books = G_reader_settings:isTrue("screensaver_exclude_finished_books") and book_finished`
The G_reader_setting things is plural, but `and book_finished` have it apply on the current book, so the local var can't be plural.

The blank line is not needed, as the whole section is about the same thing.

And I'm sure you can do shorter than adding 20 lines and you don't need that many branches.